### PR TITLE
Update modules version checker in MO to support `~=`

### PR DESCRIPTION
**Pull request by achetver**
### Details:
 - Need to support `~=` relationship operator in the "model-optimizer/mo/utils/versions_checker.py". You can find information about about this operator [here](https://www.python.org/dev/peps/pep-0440/#compatible-release). Refer to https://github.com/openvinotoolkit/openvino/pull/5171/

### Tickets:
 - 53708

Root cause analysis: Now version checker doesn't supports `~=` relationship operator for versions.

Solution: Add `~=` sign to `parse_and_filter_versions_list()` and `version_check()` functions.

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests;
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  Model Optimizer IR Reader check: N/A.

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

**Review by mvafin — APPROVED** [review]

**Comment by achetver** [comment]
@sadolini @iimironov please, take a look at these changes.

**Review by sadolini — APPROVED** [review]

**Comment by achetver** [comment]
@ Please, review this PR.

File: model-optimizer/unit_tests/mo/utils/versions_checker_test.py [hunk 1]
```diff
@@ -154,3 +165,23 @@ def test_version_check_greater(self):
         for name, key, required_version, installed_version in modules_versions_list:
             version_check(name, installed_version, required_version, key, not_satisfied_versions)
         self.assertEqual(not_satisfied_versions, ref_list)
+
+    def test_version_check_compatible(self):
+        modules_versions_list = [('module_1', '~=', '1.2.3', '1.2.3'),
```
Comment (lazarevevgeny) [inline model-optimizer/unit_tests/mo/utils/versions_checker_test.py:170]: Add unit tests provided here https://www.python.org/dev/peps/pep-0440/#compatible-release

I think that the current implementation will fail the following test: `~= 2.2.post3`

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/unit_tests/mo/utils/versions_checker_test.py (hunk 1 above)
Comment (achetver) [reply model-optimizer/unit_tests/mo/utils/versions_checker_test.py:173]: Yes, it fell. Implementation was updated and more unit tests was provided.

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**